app/parsers/bulkrax/csv_parser.rb

```diff
@@ -33,16 +33,29 @@ def import_fields
     end
 
     def required_elements?(keys)
-      return if keys.blank?
-      !required_elements.map { |el| keys.map(&:to_s).include?(el) }.include?(false)
+      return unless keys.present?
+      !missing_elements(keys).present?
+    end
+
+    def missing_elements(keys)
+      required_elements - keys.map(&:to_s)
+    end
+
+    def source_identifier
```

Comment:
if this is in application parser does it need to be here too?

Reply:
nope, removed